src/prototype_creep.js

```diff
@@ -167,9 +167,16 @@ Creep.prototype.stayInRoom = function() {
 };
 
 Creep.prototype.buildRoad = function() {
-  if (this.room.controller && this.room.controller.my && this.pos.lookFor(LOOK_TERRAIN)[0] !== 'swamp' &&
-    (this.room.controller.level < 4 || this.room.memory.misplacedSpawn)) {
-    return false;
+  if (this.room.controller && this.room.controller.my) {
+
+    if (this.pos.lookFor(LOOK_TERRAIN)[0] !== 'swamp' &&
+      (this.room.controller.level < 3 || this.room.memory.misplacedSpawn)) {
+      return false;
+    }
+    // TODO should be extracted to a method, stolen from role_harvester
+    if (!this.room.storage || this.room.storage.store.energy + 1000 < config.creep.energyFromStorageThreshold) {
```

Comment:
K, removed again. I had harvester running around and wanted to prevent that. I think the better approach is to fix the harvester pathing.

Overall I think roads bring more benefits than disadvantages. I'm improving the `upgrader` settings a bit in my workspace branch. I think we can control the storage energy level, by adapting the upgrader.